kafka-ui-react-app/src/components/Connect/List/BatchActionsBar.tsx

```diff
@@ -0,0 +1,158 @@
+  resetRowSelection(): void;
+}
+
+const BatchActionsBar: React.FC<BatchActionsBarProps> = ({
+  rows,
+  resetRowSelection,
+}) => {
+
+  const confirm = useConfirm();
+  const navigate = useNavigate();
+
+  const selectedConnectors = rows.map(({ original }) => original);
+
+  const mutationsNumber = useIsMutating();
+  const isMutating = mutationsNumber > 0;
+
+  const routerProps = useAppParams<RouterParamsClusterConnectConnector>();
+  const clusterName = routerProps.clusterName;
+  const client = useQueryClient();
+
+  const canEdit = usePermission(
+    ResourceType.CONNECT,
+    Action.EDIT,
+    routerProps.connectorName
+  );
+  const canDelete = usePermission(
+    ResourceType.CONNECT,
+    Action.DELETE,
+    routerProps.connectorName
+  );
+
+  const deleteConnectorMutation = useDeleteConnector(routerProps);
+  const deleteConnectorHandler = () =>
+    confirm(
+      <>
+        Are you sure you want to remove <b>{routerProps.connectorName}</b>{' '}
+        connector?
+      </>,
+      async () => {
+        try {
+          await deleteConnectorMutation.mutateAsync();
+          navigate(clusterConnectorsPath(clusterName));
+        } catch {
+          // do not redirect
+        }
+      }
+    );
+
+  const stateMutation = useUpdateConnectorState(routerProps);
+  const updateConnector = (action: ConnectorAction, message: string) => {
+    confirm(message, async () => {
+      try {
+        await Promise.all(
+          selectedConnectors.map((connector) => (
+            stateMutation.mutateAsync({
+              clusterName,
+              connectName: connector.connect,
+              connectorName: connector.name,
+              action,
+            })
+          ))
+        );
+        resetRowSelection();
+      } catch (e) {
+        // do nothing;
+      } finally {
+        client.invalidateQueries(['clusters', clusterName, 'connectors']);
+      }
+    });
+  };
+  const restartConnectorHandler = () => {
+    updateConnector(ConnectorAction.RESTART, 'Are you sure you want to restart selected connectors?');
+  };
+  const restartAllTasksHandler = () =>
+    updateConnector(ConnectorAction.RESTART_ALL_TASKS, 'Are you sure you want to restart all tasks in selected connectors?');
+  const restartFailedTasksHandler = () =>
+    updateConnector(ConnectorAction.RESTART_FAILED_TASKS, 'Are you sure you want to restart failed tasks in selected connectors?');
+  const pauseConnectorHandler = () =>
+    updateConnector(ConnectorAction.PAUSE, 'Are you sure you want to pause selected connectors?');
+  const resumeConnectorHandler = () =>
+    updateConnector(ConnectorAction.RESUME, 'Are you sure you want to resume selected connectors?');
+
+  return (
+    <>
+      <ActionCanButton
+        buttonSize="M"
+        buttonType="secondary"
+        onClick={pauseConnectorHandler}
+        disabled={isMutating}
+        canDoAction={canEdit}
+      >
+        Pause
+      </ActionCanButton>
+      <ActionCanButton
```

Comment:
`<ActionButton` please check this component , because the   `<ActionCanButton` is just so we would be able to use some very custom selectors together.

use `<ActionButton` and it has a `permisson` props , please check that.

so we will have a uniform way for doing the permission.

Reply:
ActionButton is only capable on checking permissions on one entity, but collection. I need either modify the way we operate permissions in ActionButton or use ActionCanButton or I can create new component ActionOnCollectionButton and implement those permissions use there. WDYT is the best way?

Reply:
I will think about it and get back to you.